src/node_file.cc

```diff
@@ -559,6 +560,9 @@ static void InternalModuleStat(const FunctionCallbackInfo<Value>& args) {
   if (rc == 0) {
     const uv_stat_t* const s = static_cast<const uv_stat_t*>(req.ptr);
     rc = !!(s->st_mode & S_IFDIR);
+    if (rc == 0) {
```

Comment:
Since this is a non hot-spot, I'd rather see an explicit `if` and `rc = 2`, similar to
```cc
if (rc == 0 && (s->st_mode & (S_IFIFO | S_IFSOCK))) {
  rc = 2;
}
```